File: services/nuldoc/nuldoc-src/pages/HomePage.ts
import GlobalFooter from "../components/GlobalFooter.ts";
import GlobalHeader from "../components/DefaultGlobalHeader.ts";
import PageLayout from "../components/PageLayout.ts";
import { Config } from "../config.ts";
import { elem, Element } from "../dom.ts";

export default async function HomePage(config: Config): Promise<Element> {
  return await PageLayout({
    metaCopyrightYear: config.site.copyrightYear,
    metaDescription: "nsfisis のサイト",
    metaTitle: config.sites.default.siteName,
    metaAtomFeedHref: `https://${config.sites.default.fqdn}/atom.xml`,
    site: "default",
    config,
    children: elem(
      "body",
      { class: "single" },
      GlobalHeader({ config }),
      elem(
        "main",
        { class: "main" },
        elem(
          "article",
          { class: "post-single" },
          elem(
            "article",
            { class: "post-entry" },
            elem(
              "a",
              { href: `https://${config.sites.about.fqdn}/` },
              elem(
                "header",
                { class: "entry-header" },
                elem("h2", {}, "About"),
              ),
            ),
          ),
          elem(
            "article",
            { class: "post-entry" },
            elem(
              "a",
              { href: `https://${config.sites.blog.fqdn}/posts/` },
              elem("header", { class: "entry-header" }, elem("h2", {}, "Blog")),
            ),
          ),
          elem(
            "article",
            { class: "post-entry" },
            elem(
              "a",
              { href: `https://${config.sites.slides.fqdn}/slides/` },
              elem(
                "header",
                { class: "entry-header" },
                elem("h2", {}, "Slides"),
              ),
            ),
          ),
          elem(
            "article",
            { class: "post-entry" },
            elem(
              "a",
              { href: `https://repos.${config.sites.default.fqdn}/` },
              elem(
                "header",
                { class: "entry-header" },
                elem("h2", {}, "Repositories"),
              ),
            ),
          ),
        ),
      ),
      GlobalFooter({ config }),
    ),
  });
}

File: services/nuldoc/lib/nuldoc/pages/home_page.rb
module Nuldoc
  module Pages
    class HomePage
      def initialize(config:)
        @config = config
      end

      def render
        config = @config
        Components::PageLayout.new(
          meta_copyright_year: config.site.copyright_year,
          meta_description: 'nsfisis のサイト',
          meta_title: config.sites.default.site_name,
          meta_atom_feed_href: "https://#{config.sites.default.fqdn}/atom.xml",
          site: 'default',
          config: config,
          children: DOM::HTMLBuilder.new.build do
            body class: 'single' do
              render(Components::DefaultGlobalHeader, config: config)
              main class: 'main' do
                article class: 'post-single' do
                  article class: 'post-entry' do
                    a href: "https://#{config.sites.about.fqdn}/" do
                      header(class: 'entry-header') { h2 { text 'About' } }
                    end
                  end
                  article class: 'post-entry' do
                    a href: "https://#{config.sites.blog.fqdn}/posts/" do
                      header(class: 'entry-header') { h2 { text 'Blog' } }
                    end
                  end
                  article class: 'post-entry' do
                    a href: "https://#{config.sites.slides.fqdn}/slides/" do
                      header(class: 'entry-header') { h2 { text 'Slides' } }
                    end
                  end
                  article class: 'post-entry' do
                    a href: "https://repos.#{config.sites.default.fqdn}/" do
                      header(class: 'entry-header') { h2 { text 'Repositories' } }
                    end
                  end
                end
              end
              render(Components::GlobalFooter, config: config)
            end
          end
        ).build
      end
    end
  end
end
